File: dksim/view/shared/fight_time_variance.py
#Roll for time variance
import random
from dksim.view.shared.damage_array_updater import damage_array_updater
#, standard_10k_random_value, damage_result_number
def time_variance_rolled(fight_variance, fight_length, standard_10k_random_value, damage_result_number):
    add_or_sub = random.randint(0, 1)
    amount_to_add_or_sub = (random.randint(0, (fight_variance * 4)) / 4)
    #add_or_sub = standard_10k_random_value[damage_result_number]/10000
    if add_or_sub > .5:
        add_or_sub = 1
    else:
        add_or_sub = 0
    #damage_result_number = damage_array_updater(damage_result_number)
    #amount_to_add_or_sub = fight_variance*(standard_10k_random_value[damage_result_number]/10000)
    #damage_result_number = damage_array_updater(damage_result_number)
    if add_or_sub == 0:
        return_variance = fight_length + amount_to_add_or_sub
    elif add_or_sub == 1:
        return_variance = fight_length - amount_to_add_or_sub
    return return_variance, damage_result_number

File: dksim/view/shared/fight_time_variance.cpp
#include <iostream>
#include <random>

int time_variance_rolled(int fight_variance, int fight_length){
    // Seed the random number engine with a random device
    std::random_device rd;
    std::mt19937 rng(rd());

    // Define the distribution for the desired range
    std::uniform_int_distribution<int> distribution(0, 1);
    // Generate a random number between the low damage int and the high damage int
    int add_or_sub = distribution(rng);

    // Seed the random number engine with a random device
    std::random_device rdt;
    std::mt19937 rngt(rdt());

    // Define the distribution for the desired range
    std::uniform_int_distribution<int> distributiont(0, fight_variance);
    // Generate a random number between the low damage int and the high damage int
    int amount_to_add_or_sub = distributiont(rngt);

    int return_variance = 0;

    if (add_or_sub == 0){
        return_variance = fight_length + amount_to_add_or_sub;
    }
    else {
        return_variance = fight_length - amount_to_add_or_sub;
    }

    return return_variance;
}